src/providers/stripe_sca.js

```diff
@@ -0,0 +1,222 @@
+                <label class="multiple_card_tokenization__hosted-fields--label" for="cardholder-name_${postfix}">Cardholder Name</label>
+                <input type="text" id="cardholder-name_${postfix}" class="multiple_card_tokenization__hosted-field" placeholder="CARDHOLDER NAME" />
+              </div>
+              <div class="multiple_card_tokenization__field-container">
+                <label class="multiple_card_tokenization__hosted-fields--label">Card Data</label>
+                <div id="card-element"></div>
+                <div id="card-errors" role="alert"></div>
+              </div>
+            </div>
+
+            ${submit_button}
+          </form>
+        <div>
+      </div>
+    </div>
+  `;
+
+  if (!target) {body.appendChild(div);}
+
+  modal = document.getElementById(`modal_${postfix}`);
+}
+
+function _checkLoading() {
+  if (window.Stripe) {
+    _initializeScripts();
+    clearInterval(loadingInterval);
+  }
+}
+
+function _initializeScripts() {
+  const { postfix, connection, showSubmitButton } = gatewaySettings;
+  const { token } = connection;
+
+  form = document.querySelector(`#stripe_card_form_${postfix}`);
+
+  if (showSubmitButton === undefined || !showSubmitButton === false) {
+    submit = document.querySelector(`#submit_${postfix}`);
+    closeBTN = document.querySelector(`#close-form-${postfix}`);
+    closeBTN.addEventListener('click', hideForm.bind(this), false);
+  }
+
+  form.addEventListener('submit', onSubmit.bind(this), false);
+  stripe = Stripe(token);
+  elements = stripe.elements();
+
+  var style = {
+    base: {
+      marginTop: '5px',
+      fontSize: '14px',
+      lineHeight: '24px'
+    }
+  };
+
+  var cardSettings = {
+    hidePostalCode: true
+  };
+
+  card = elements.create('card', {style: style, card: cardSettings});
+  card.mount('#card-element');
+  card.addEventListener('change', function(event) {
+    var displayError = document.getElementById('card-errors');
+    if (event.error) {
+      displayError.textContent = event.error.message;
+    } else {
+      displayError.textContent = '';
+    }
+  });
+}
+
+function preparePaymentIntentForBooking(bookingToken, paymentMethodId) {
+  return fetch(`${DOMAIN}/api/public/v1/prepare_online_payment/stripe_sca?token=${bookingToken}`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      payment_method_id: paymentMethodId
+    })
+  });
+}
+
+function prepareIntent(amount, apiKey, paymentMethodId) {
+  return fetch(`${DOMAIN}/api/public/v1/prepare_online_payment/stripe_sca`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      'Token': apiKey,
+    },
+    body: JSON.stringify({
+      amount: amount,
+      payment_method_id: paymentMethodId
+    })
+  });
+}
+
+function onSubmit(event) {
+  event.preventDefault();
+
+  stripe.createPaymentMethod('card', card).then(function(result) {
```

Comment:
need to check what billing information contains payment method and what information we have on form